display-points-using-clustering-feature-reduction/src/main/java/com/esri/arcgismaps/sample/displaypointsusingclusteringfeaturereduction/components/MapViewModel.kt

```diff
@@ -0,0 +1,232 @@
+    val clusterInfoList = mutableStateListOf<AnnotatedString>()
+
+    // the title of the popup result
+    val popupTitle = mutableStateOf("")
+
+    init {
+        // show loading dialog to indicate that the map is loading
+        showLoadingDialog.value = true
+        // load the portal and create a map from the portal item
+        val portalItem = PortalItem(
+            Portal(application.getString(R.string.portal_url)),
+            "8916d50c44c746c1aafae001552bad23"
+        )
+
+        // set the map to be displayed in the layout's MapView
+        mapViewState.value.arcGISMap = ArcGISMap(portalItem)
+
+        sampleCoroutineScope.launch {
+            mapViewState.value.arcGISMap.load().onSuccess {
+                showLoadingDialog.value = false
+            }
+        }
+    }
+
+    /**
+    `* Toggle the FeatureLayer's featureReduction property
+     */
+    fun toggleFeatureReduction() {
+        val map = mapViewState.value.arcGISMap
+        isFeatureReductionEnabled.value = !isFeatureReductionEnabled.value
+        if (map.loadStatus.value == LoadStatus.Loaded) {
+            map.operationalLayers.forEach { layer ->
+                when (layer) {
+                    is FeatureLayer -> {
+                        layer.featureReduction?.isEnabled = isFeatureReductionEnabled.value
+                    }
+
+                    else -> {}
+                }
+            }
+        }
+    }
+
+    /**
+     * Identify the feature layer results and display the resulting popup element information
+     */
+    fun handleIdentifyResult(result: Result<List<IdentifyLayerResult>>) {
+        sampleCoroutineScope.launch {
+            result.onSuccess { identifyResultList ->
+                // initialize the string for each tap event resulting in a new identifyResultList
+                clusterInfoList.clear()
+                popupTitle.value = ""
+                identifyResultList.forEach { identifyLayerResult ->
+                    val popups = identifyLayerResult.popups
+                    popups.forEach { popup ->
+                        // set the popup title
+                        popupTitle.value = popup.title
+                        // show the bottom sheet for the popup content
+                        showClusterSummaryBottomSheet.value = true
+                        popup.evaluateExpressions().onSuccess {
+                            popup.evaluatedElements.forEach { popupElement ->
+                                when (popupElement) {
+                                    is FieldsPopupElement -> {
+                                        popupElement.fields.forEach { popupField ->
+                                            // convert popupField.label embedded with html tags using HtmlCompat.fromHtml
+                                            clusterInfoList.add(
+                                                HtmlCompat.fromHtml(
+                                                    popupField.label,
```

Comment:
Usually, for a field in a field popup element popup shows both the label and value of the field instead of just the label. e.g. `population: 11,100`.

Reply:
Done, updated